multimodal/src/autogluon/multimodal/data/utils.py

```diff
@@ -425,3 +443,97 @@ def get_text_token_max_len(provided_max_len, config, tokenizer, checkpoint_name)
         max_len = min(provided_max_len, default_max_len)
 
     return max_len
+
+
+def construct_processor(
+    size,
+    normalization,
+    transform_types: List[str],
+) -> transforms.Compose:
+    """
+    Build up an image processor from the provided list of transform types.
+
+    Parameters
+    ----------
+    transform_types
+        A list of image transform types.
+
+    Returns
+    -------
+    A torchvision transform.
+    """
+    processor = []
+    for trans_type in transform_types:
+        args = None
+        kargs = None
+        if "(" in trans_type:
+            trans_mode = trans_type[0 : trans_type.find("(")]
+            if "{" in trans_type:
+                kargs = ast.literal_eval(trans_type[trans_type.find("{") : trans_type.rfind(")")])
+            else:
+                args = ast.literal_eval(trans_type[trans_type.find("(") :])
+        else:
+            trans_mode = trans_type
+
+        if trans_mode == "resize_to_square":
+            processor.append(transforms.Resize((size, size), interpolation=BICUBIC))
+        elif trans_mode == "resize_shorter_side":
+            processor.append(transforms.Resize(size, interpolation=BICUBIC))
+        elif trans_mode == "center_crop":
+            processor.append(transforms.CenterCrop(size))
+        elif trans_mode == "horizontal_flip":
+            processor.append(transforms.RandomHorizontalFlip())
+        elif trans_mode == "vertical_flip":
+            processor.append(transforms.RandomVerticalFlip())
+        elif trans_mode == "color_jitter":
+            if kargs is not None:
+                processor.append(transforms.ColorJitter(**kargs))
+            elif args is not None:
+                processor.append(transforms.ColorJitter(*args))
+            else:
+                processor.append(transforms.ColorJitter(brightness=0.1, contrast=0.1, saturation=0.1))
+        elif trans_mode == "affine":
+            if kargs is not None:
+                processor.append(transforms.RandomAffine(**kargs))
+            elif args is not None:
+                processor.append(transforms.RandomAffine(*args))
+            else:
+                processor.append(transforms.RandomAffine(degrees=15, translate=(0.1, 0.1), scale=(0.9, 1.1)))
+        elif trans_mode == "randaug":
+            if kargs is not None:
+                processor.append(RandAugment(**kargs))
+            elif args is not None:
+                processor.append(RandAugment(*args))
+            else:
+                processor.append(RandAugment(2, 9))
+        elif trans_mode == "trivial_augment":
+            processor.append(TrivialAugment(IMAGE, 30))
+        else:
+            raise ValueError(f"unknown transform type: {trans_mode}")
+
+    processor.append(transforms.ToTensor())
+    processor.append(normalization)
+    return transforms.Compose(processor)
+
+
+def mean_std(norm_type: str):
```

Comment:
Same as above, can't tell this is for image modality from function name.

Reply:
addressed